airflow-core/docs/howto/deadline-alerts.rst

```diff
@@ -0,0 +1,312 @@
+    after it was scheduled to start, regardless of when (or if) it actually began executing.
+    Useful for ensuring scheduled Dags complete before their next scheduled run.
+
+``DeadlineReference.FIXED_DATETIME``
+    Specifies a fixed point in time. Useful when Dags must complete by a specific time.
+
+Here's an example using a fixed datetime:
+
+.. code-block:: python
+
+    tomorrow_at_ten = datetime.combine(datetime.now().date() + timedelta(days=1), time(10, 0))
+
+    with DAG(
+        dag_id="fixed_deadline_alert",
+        deadline=DeadlineAlert(
+            reference=DeadlineReference.FIXED_DATETIME(tomorrow_at_ten),
+            interval=timedelta(minutes=-30),  # Alert 30 minutes before the reference.
+            callback=AsyncCallback(
+                SlackWebhookNotifier,
+                kwargs={
+                    "slack_conn_id": "slack_default",
+                    "channel": "#alerts",
+                    "text": "Dag 'slack_deadline_alert' still running after 30 minutes.",
+                    "username": "Airflow Alerts",
+                },
+            ),
+        ),
+    ):
+        EmptyOperator(task_id="example_task")
+
+The timeline for this example would look like this:
+
+::
+
+    |------|----------|---------|------------|--------|
+         Queued     Start    Deadline    Reference
+         09:15      09:17     09:30       10:00
+
+.. note::
+    Note that since the interval is a negative value, the deadline is before the reference in this case.
+
+Using Callbacks
+---------------
+
+When a deadline is exceeded, the callback is executed. You can use an existing :doc:`Notifier </howto/notifications>`
+or create a custom callback function.  A callback must be either an :class:`~airflow.sdk.definitions.deadline.AsyncCallback`
+or a :class:`~airflow.sdk.definitions.deadline.SyncCallback`.
+
+Using Built-in Notifiers
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here's an example using the Slack Notifier if the Dag run has not finished within 30 minutes of it being queued:
+
+.. code-block:: python
+
+    with DAG(
+        dag_id="slack_deadline_alert",
+        deadline=DeadlineAlert(
+            reference=DeadlineReference.DAGRUN_QUEUED_AT,
+            interval=timedelta(minutes=30),
+            callback=AsyncCallback(
+                SlackWebhookNotifier,
+                kwargs={
+                    "slack_conn_id": "slack_default",
+                    "channel": "#alerts",
+                    "text": "Dag 'slack_deadline_alert' still running after 30 minutes.",
+                    "username": "Airflow Alerts",
+                },
+            ),
+        ),
+    ):
+        EmptyOperator(task_id="example_task")
+
+Creating Custom Callbacks
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can create custom callbacks for more complex handling. The ``kwargs`` specified in the ``Callback``
+are passed to the callback function, if any are provided.  **Synchronous callbacks** (standard python
```

Comment:
```suggestion
are passed to the callback function, if any are provided. **Synchronous callbacks** (standard python
```

Reply:
Hey, stop picking on my age!  Double-spacer for life!  ((I'll change them))